src/main/java/teammates/ui/pagedata/InstructorFeedbackResultsPageData.java

```diff
@@ -1178,6 +1160,14 @@ private void buildTableColumnHeaderForRecipientQuestionGiverView(List<ElementTag
 
             configureResponseRow(response.giver, response.recipient, responseRow);
 
+            if (isFirstGroupedByGiver && question.getQuestionDetails().isCommentsOnResponsesAllowed()) {
+                List<String> responseGiverRecipientList = new ArrayList<>(bundle.emailNameTable.keySet());
```

Comment:
If we use map for the indexes, we can put the logic when `question.giverType == FeedbackParticipantType.TEAMS` to here.

Reply:
I think it may not work. Can you please clarify what you mean?

Reply:
```java
if (question.giverType == FeedbackParticipantType.TEAMS && bundle.getRoster().isStudentInCourse(response.giver)) {
    // when giver type is team while a student's email is stored, bla bla 
    responseGiverRecipientIndexes.put(response.giver, responseGiverRecipientIndexes.get(response.giver + Const.TEAM_OF_EMAIL_OWNER))
}
```

I just notice that we shall also do this for recipient?

Reply:
okay. thanks for the clarification.